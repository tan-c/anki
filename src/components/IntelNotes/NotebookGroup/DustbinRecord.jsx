import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { UiActions } from 'utility-redux/ui';
import { NoteActions } from 'utility-redux/note';

export class DustbinRecord extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const {
      activeNotebookId, notes,
    } = this.props;

    return (
      <div
        data-role="dustbin-record"
        role="menuitem" tabIndex="-1" className={`flex-container-row typical-setup ${activeNotebookId === 'deleted' && 'bg-red'}`} onClick={(_) => { this.props.UiActions.updateIn(['himalayan', 'activeNotebookId'], 'deleted'); }}
      >
        <i className="fa fa-fw fa-trash-o wdith-20" />
        <div className="flex-1">Dustbin</div>
        <span>{notes.filter(note => note.get('isDeleted')).size}</span>
        <i
          className="fa fa-fw fa-trash-o wdith-20" role="button" tabIndex="-1" onClick={(event) => {
            if (confirm('Clear dustbin?')) {
              this.props.NoteActions.deleteRecords({ isDeleted: true });
            }
          }}
        />
      </div>
    );
  }
}

DustbinRecord.defaultProps = {
  activeNotebookId: '',
};

DustbinRecord.propTypes = {
  notes: PropTypes.object.isRequired,
  activeNotebookId: PropTypes.string,

  NoteActions: PropTypes.object.isRequired,
  UiActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    notes: state.notes,
    activeNotebookId: state.ui.getIn(['himalayan', 'activeNotebookId']),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    NoteActions: bindActionCreators(NoteActions, dispatch),
    UiActions: bindActionCreators(UiActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(DustbinRecord);
