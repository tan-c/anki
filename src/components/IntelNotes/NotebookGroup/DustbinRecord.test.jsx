import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

import { Map } from 'immutable';

import { DustbinRecord } from './DustbinRecord';

function setup() {
  const props = {
    notes: Map(),
    activeNotebookId: '',

    NoteActions: {},
    UiActions: {},
  };

  return shallow(<DustbinRecord {...props} />);
}

describe('DustbinRecord Unit Tests', () => {
  describe('', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      // expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("div[data-role='dustbin-record']")).toHaveLength(1);
    });
  });
});
