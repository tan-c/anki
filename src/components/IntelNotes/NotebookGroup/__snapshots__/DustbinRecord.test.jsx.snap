// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DustbinRecord Unit Tests  renders correctly 1`] = `
<div
  className="flex-container-row typical-setup false"
  data-role="dustbin-record"
  onClick={[Function]}
  role="menuitem"
  tabIndex="-1"
>
  <i
    className="fa fa-fw fa-trash-o wdith-20"
  />
  <div
    className="flex-1"
  >
    Dustbin
  </div>
  <span>
    0
  </span>
  <i
    className="fa fa-fw fa-trash-o wdith-20"
    onClick={[Function]}
    role="button"
    tabIndex="-1"
  />
</div>
`;
